x-pack/legacy/plugins/lens/index.ts

```diff
@@ -77,6 +81,19 @@ export const lens: LegacyPluginInitializer = kibana => {
           },
         },
       });
+
+      // Set up with the new platform plugin lifecycle API.
+      const plugin = lensServerPlugin();
+      await plugin.setup(({
+        http: {
+          ...kbnServer.newPlatform.setup.core.http,
+          createRouter: () => kbnServer.newPlatform.setup.core.http.createRouter('/api/lens'),
+        },
+      } as unknown) as CoreSetup);
+
+      server.events.on('stop', async () => {
```

Comment:
You're right, I'm going to make it all synchronous. I was going back and forth on this while developing because I think we will eventually want async here once we are fully on the new platform.